GOAT-Compiler/TypeChecker.cs

```diff
@@ -0,0 +1,406 @@
+        {
+            _typeDictionary.Add(node, numberType(node.GetNumber().Text));
+        }
+
+        public override void OutAPlusExp(APlusExp node)
+        {
+            ExpressionTypeChecker(node.GetL(), node.GetR(), node);
+        }
+
+        public override void OutAMinusExp(AMinusExp node)
+        {
+            ExpressionTypeChecker(node.GetL(), node.GetR(), node);
+        }
+
+        public override void OutABoolvalExp(ABoolvalExp node)
+        {
+            _typeDictionary.Add(node, Types.Boolean);
+        }
+
+        public override void OutAAndExp(AAndExp node)
+        {
+            AndAndOr(node.GetL(), node.GetR(), node);
+        }
+        public override void OutAOrExp(AOrExp node)
+        {
+            AndAndOr(node.GetL(), node.GetR(), node);
+        }
+        public override void OutAEqExp(AEqExp node)
+        {
+            EqelNotEqel(node.GetL(), node.GetR(), node);
+        }
+
+        private void MultDivMod(Node left, Node right, Node current)
+        {
+            Types leftType = _typeDictionary[left];
+            Types rightType = _typeDictionary[right];
+            if ((leftType == Types.Integer || leftType == Types.FloatingPoint) && (rightType == Types.Integer || rightType == Types.FloatingPoint))
+            {
+                ExpressionTypeChecker(left, right, current);
+            }
+            else
+            {
+                throw new TypeMismatchException(current);
+            }
+        }
+        public override void OutAModuloExp(AModuloExp node)
+        {
+            MultDivMod(node.GetL(), node.GetL(), node);
+        }
+
+        public override void OutAMultExp(AMultExp node)
+        {
+            MultDivMod(node.GetL(), node.GetL(), node);
+        }
+
+        public override void OutADivdExp(ADivdExp node)
+        {
+            MultDivMod(node.GetL(), node.GetL(), node);
+        }
+        public override void OutAGeqExp(AGeqExp node)
+        {
+            GreaterThanLessThan(node.GetL(), node.GetR(), node);
+        }
+
+        public override void OutAGtExp(AGtExp node)
+        {
+            GreaterThanLessThan(node.GetL(), node.GetR(), node);
+        }
+        public override void OutALtExp(ALtExp node)
+        {
+            GreaterThanLessThan(node.GetL(), node.GetR(), node);
+        }
+        public override void OutALeqExp(ALeqExp node)
+        {
+            GreaterThanLessThan(node.GetL(), node.GetR(), node);
+        }
+        public override void OutANeqExp(ANeqExp node)
+        {
+            EqelNotEqel(node.GetL(), node.GetR(), node);
+        }
+
+        public override void OutAAssignStmt(AAssignStmt node)
+        {
+            Symbol id = _symbolTable.GetSymbol(node.GetId().Text);
+            if (Convert(node.GetExp(), id.type) == id.type)
+            {
+                _typeDictionary.Add(node, id.type);
+            }
+            else
+            {
+                throw new TypeMismatchException(node);
```

Comment:
You already throw this exception in convert...

Reply:
Removed